jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -348,6 +500,49 @@ public Number getParam() {
     public String getManagerHostPort() {
       return managerHostPort;
     }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public static class Builder {
+      private String type;
+      private Number param;
+      private String managerHostPort;
+
+      public Builder builder() {
+        return new Builder();
+      }
+
+      /**
+       * Set the sampler type
+       *
+       * @see ConstSampler#TYPE
+       * @see ProbabilisticSampler#TYPE
+       * @see RateLimitingSampler#TYPE
+       * @see RemoteControlledSampler#TYPE
+       * @param type the sampler type
+       * @return this builder
+       */
+      public Builder withType(String type) {
```

Comment:
Do we need `withType` and `withParam` as programmatic API? It's easier to directly consume `withSampler(Sampler)`.

Reply:
I added a new `withSampler`, which resets any value that has been set for type/param/managerHostPort (and vice-versa: setting type/param/managerHostPort resets a previously given Sampler)